Add ebill parser

AllanWang opened the pull request:
Old version is [here](https://github.com/jguerinet/MyMartlet/blob/e73e265f08567600024ffdd25032f7d1f6c7541e/app/src/main/java/util/retrofit/EbillConverter.kt#L58)

The logic is generally simplified. For each row, I try to fetch a list of dates and an amount. If I get two dates and an amount, then I create a statement. I will map all rows to nonnull statements and return the final result

jguerinet commented:
* Did we not decided on `@property` instead of `@param`?
* I just checked and spotless is configured to end everything with a new line, but that doesn't seem to be the case in the new files you've added. Any idea what happened there?

AllanWang commented:
Where did you check within spotless? As for `@param`, I guess I didn't properly convert everything. The new class is copied form the app code though, and it should have been switched there.

AllanWang commented:
Just a note here that I've added test runs to travis, added auto cancellation, and disable pr builds (through travis settings online). I think there's still throttling with open source projects, and three quick commits end up triggering 6 builds, which are more or less sequential. I'd rather wait for one build on the newest commit only if others haven't started

nedtwigg commented on parser/src/main/kotlin/com/guerinet/mymartlet/model/Statement.kt:
```diff
@@ -25,4 +25,4 @@ data class Statement(
     val date: LocalDate,
     val dueDate: LocalDate,
     val amount: Double
-)
```
Re: https://github.com/diffplug/spotless/issues/359

Note the "no-newline" on the left, while it's missing on the right.  That's because the file on the left is missing a newline, but the file on the right has it.

AllanWang replied:
Hm you're right. I think this was a false alarm, as we thought no new lines existed from the github preview. They all exist locally, so I will close that issue

jguerinet commented on parser/src/main/kotlin/com/guerinet/mymartlet/parser/General.kt:
```diff
@@ -69,14 +81,13 @@ fun String.parseDate(year: Int): LocalDate {
  */
-fun String.parseDateRange(year: Int): Pair<LocalDate, LocalDate> {
+fun String.parseDateMDRange(year: Int): Pair<LocalDate, LocalDate>? {
     //Split the range into the 2 date Strings
```
Space missing between `//` and beginning of comment.
